src/map/WorldMap.kt

```diff
@@ -42,9 +17,38 @@ interface WorldMap {
     fun positionFor(mapObject: MapObject): Coordinate
     fun getNode(positionFor: Coordinate): Node
     fun moveObject(actor: MapObject, vector: Vector)
+    fun getAllCoordinates(): Set<Coordinate>
+    fun getSurrounding(actor: Actor): WorldMap
 }
 
-class CoordinateNodeWorldMap(private val nodes: MutableMap<Coordinate, Node>) : WorldMap {
+class Memory(private val nodes: MutableMap<Coordinate, Node>) {
```

Comment:
This is specific to certain Player/family of players. It should not be in the api/library class.

Reply:
ok moved it in the player class which is called Bradley